### PR TITLE
Reworked nonRootUser use

### PR DESCRIPTION
**Pull request by litong01**
This PR is to fix the issues deploy astra connector/neptune on openshift cluster.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Build the manager binary
-FROM golang:1.20.5 as builder
+FROM --platform=${BUILDPLATFORM} golang:1.19 as builder
```
Comment (litong01) [inline Dockerfile:2]: This actually fixed the issue for multi-platform build which was broken.

Comment (oscarr-ntap) [reply Dockerfile:2]: --platform=${BUILDPLATFORM} makes sense but we want 1.20 or 1.21 goLang version

Comment (litong01) [reply Dockerfile:2]: @oscarr-ntap will change to the 1.21.

**Review by oscarr-ntap — APPROVED** [review]
Looks good but neptune.yaml is generated so merge https://github.com/NetApp-Polaris/neptune/pull/542/files first.

**Pull request merged**